File: Time Converter/Time Converter.cpp
#include <iostream>

using namespace std;

void TimeConversion() {
    string ttype;
    double min, result, hours, days, weeks, months, years;
    cout << "Type (s), (m), (h), (d), (w), (M), (y) for second, minute, hour, day, week, month and year conversion." << endl;
    cin >> ttype;

    switch (ttype[0]) {
        case 's':
            cout << "How many minutes?" << endl;
            cin >> min;
            result = min * 60;
            cout << result << " seconds" << endl;
            break;
        case 'm':
            cout << "How many hours?" << endl;
            cin >> hours;
            result = hours * 60;
            cout << result << " minutes" << endl;
            break;
        case 'h':
            cout << "How many days?" << endl;
            cin >> days;
            result = days * 24;
            cout << result << " hours" << endl;
            break;
        case 'd':
            cout << "How many weeks?" << endl;
            cin >> weeks;
            result = weeks * 7;
            cout << result << " days" << endl;
            break;
        case 'w':
            cout << "How many months?" << endl;
            cin >> months;
            result = months * 4.34812141;
            cout << result << " weeks" << endl;
            break;
        case 'M':
            cout << "How many years?" << endl;
            cin >> years;
            result = years * 12.25;
            cout << result << " months" << endl;
            break;
        case 'y':
            cout << "How many hours?" << endl;
            cin >> hours;
            result = hours / 8766;
            cout << result << " years" << endl;
            break;
        default:
            cout << "Invalid input." << endl;
            break;
    }
}

int TimeConversionMain() {
    TimeConversion();
    return 0;
}

File: Time convertion/timeconverter.go
package main

import (
	"fmt"
)

func main() {
	var ttype string
	var min, result, hours, days, weeks, months, years float64
	fmt.Println("Type (s), (m), (h), (d), (w), (M), (y) for second, minute, hour, day, week, month and year convertion.")
	fmt.Scan(&ttype)

	switch ttype {
	case "s":
		fmt.Println("How many minutes?")
		fmt.Scan(&min)
		result = min * 60
		fmt.Printf("%.2f seconds", result)
	case "m":
		fmt.Println("How many hours?")
		fmt.Scan(&hours)
		result = hours * 60
		fmt.Printf("%.2f minutes", result)
	case "h":
		fmt.Println("How many days?")
		fmt.Scan(&days)
		result = days * 24
		fmt.Printf("%.2f hours", result)
	case "d":
		fmt.Println("How many weeks?")
		fmt.Scan(&weeks)
		result = weeks * 7
		fmt.Printf("%.2f days", result)
	case "w":
		fmt.Println("How many months?")
		fmt.Scan(&months)
		result = months * 4.34812141
		fmt.Printf("%.2f weeks", result)
	case "M":
		fmt.Println("How many years?")
		fmt.Scan(&years)
		result = years * 12.25
		fmt.Printf("%.2f months", result)
	case "y":
		fmt.Println("How many hours?")
		fmt.Scan(&hours)
		result = hours / 8766
		fmt.Printf("%f years", result)
	}
}
